scipy/optimize/_minimize.py

```diff
@@ -84,7 +84,7 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
               see below for description.
 
         If not given, chosen to be one of ``BFGS``, ``L-BFGS-B``, ``SLSQP``,
-        depending if the problem has constraints or bounds.
+        depending on if the problem has constraints or bounds.
```

Comment:
Not a native speaker but it does not seem more correct. Better would be `depending on whether or not`

Reply:
"depend" is intransitive, so it could not take objects directly. And "whether or not" is essentially the same as "if." I can change it as you proposed, but having two "or" in the clause might be a little confusing.

Reply:
What I mean is that I would have wrote `depending on whether or not the problem...`

Reply:
What I mean is that I would have wrote `depending on whether or not the problem...`

Reply:
Yeah I understand. If `depending on whether or not the problem has constraints or bounds` is clearer to you I will make it so~

Reply:
Modified https://github.com/scipy/scipy/pull/14720/commits/ab246c8cea76ccaf02778736d23b1f5a824694fc